Objects/typeobject.c

```diff
@@ -7240,6 +7240,11 @@ update_slot(PyTypeObject *type, PyObject *name)
        recursing into subclasses. */
     PyType_Modified(type);
 
+    if (PyUnicode_READ_CHAR(name, 0) != '_') {
```

Comment:
Could you check this before calling `update_slot()`? In `update_all_slots()` it is called with all dunder names.

It may be worth to perform more strict check and speed up also setting protected and private attributes.

Reply:
Probably yes, preferably in a little helper function in that case, like `looks_like_special_name()`.
But wouldn't we still have to call `PyType_Modified()`?

Reply:
I implemented a tighter check. Timings are pretty much the same.